### PR TITLE
Update how Windows ZIP artefacts will be managed

### PR DESCRIPTION
**Pull request by asgrim**
After discussion with @shivammathur and @derickr we are tweaking the requirements around how the Windows binaries will be structured. Key differences:

 * Windows artefacts attached to GH releases will be a .zip file instead of a .dll
 * How the contents of the .zip file will be extracted has been defined

### CODE REVIEW

**Review by shivammathur — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -215,6 +215,15 @@ The descriptions of these items:
  * `-nts` - optional - non-thread safe. If omitted, thread-safe mode (ZTS) is assumed.
  * `-arch` - for example `x86_64`, fetch using `php -r "echo php_uname('m');"`.
 
+#### Contents of the Windows ZIP
+
+The pre-built ZIP should contain at minimum a DLL named `php_{extension-name}.dll`. It may include additional resources,
```
Comment (derickr) [inline README.md:220]: I would really like to see `php_{extension-name}-{tag}-{php-maj/min}-{compiler}{"-nts"}?{-arch}.dll` as filenames. and not just `php_{extension-name}.dll`, as having multiple `php_{extension-name}.dll` files makes it really hard to see which is which.

File: README.md [hunk 2]
```diff
@@ -273,7 +282,18 @@ $ pie install xdebug --not-a-defined-configuration-option # this would fail
 
 ### Windows installation
 
-Determine the expected name for the Windows DLL:
+When installing the extension on Windows, the PHP version invoking PIE would be used, unless `--with-php-path=<php-path>` is
```
Comment (derickr) [inline README.md:285]: "would be used as path to check the needed version, and install path" maybe? It's not going to be used for *running* PIE.

File: README.md [hunk 3]
```diff
@@ -273,7 +282,18 @@ $ pie install xdebug --not-a-defined-configuration-option # this would fail
...
+provided. Example:
+
+```powershell
+# Assuming C:\usr\php8.3.4\php.exe is in the $PATH
+$ pie install xdebug                                     # uses C:\usr\php8.3.4\php.exe (i.e. the version that invoked PIE)
+$ pie install xdebug --with-php-path="C:\usr\php7.4.33"  # uses C:\usr\php7.4.33\php.exe
```
Comment (derickr) [inline README.md:291]: nit: These lines are too long, they even wrap in the GH review pane.

File: README.md (hunk 1 above)
Comment (asgrim) [reply README.md:220]: This kinda goes against the convention of how the .dlls are already named, I believe? Once they're in situ, the php version is usually going to be inferred by the path, or maybe there is only one PHP path on the system, e.g. `C:\php\php8.1.2\ext\php_xdebug.dll` - it's not a dealbreaker to change this to use the "detailed name" as you suggested, but I wonder if it's superfluous? :thinking:

Comment (derickr) [reply README.md:220]: I was only talking about the contents of the zip file. I agree they should be *installed* as `php_{extension_name}.dll`.

Comment (asgrim) [reply README.md:220]: Understood, yes I think that's acceptable :+1:

File: README.md (hunk 3 above)
Comment (asgrim) [reply README.md:291]: Will reflow the whole MD doc to 80 characters after merge :+1:

File: README.md (hunk 1 above)
Comment (asgrim) [reply README.md]: Fixed in 1d94de9762a5145dc47ea97385355cb7dc0c186e - please see that update @shivammathur for when the package is produced, the included `.dll` naming should match the `.zip`

File: README.md (hunk 2 above)
Comment (asgrim) [reply README.md]: Fixed, thank you!

File: README.md (hunk 1 above)
Comment (shivammathur) [reply README.md]: Noted 👍. I will make the change

**Pull request merged**